src/Adapter/Cart/CommandHandler/AddProductToCartHandler.php

```diff
@@ -66,27 +67,31 @@ public function __construct(
      */
     public function handle(AddProductToCartCommand $command): void
     {
-        $quantity = $command->getQuantity();
-        $this->assertQuantityIsPositiveInt($quantity);
-
         $cartIdValue = $command->getCartId()->getValue();
         $productIdValue = $command->getProductId()->getValue();
-        $combinationId = $command->getCombinationId();
+        $combinationId = $command->getCombinationId() ? $command->getCombinationId()->getValue() : null;
```

Comment:
```suggestion
        $combinationId = $command->getCombinationId() instanceof CombinationId ? $command->getCombinationId()->getValue() : null;
```

Reply:
:+1:

Reply:
@Arman-Hosseini As `getCombinationId()` is returning a `CombinationId` (explicitly said by the method signature) I don't see why I should check for that? I could understand being more strict by checking for `null` though. Am I missing something? 🙂

Reply:
Agree with you @atomiix, but maybe better to be explicit here and check for not `null` 
```suggestion
        $combinationId = $command->getCombinationId() !== null ? $command->getCombinationId()->getValue() : null;
```
From:
```
public function getCombinationId(): ?CombinationId
```

Reply:
@atomiix For clarity it's better to compare it with the type that ultimately returns.

Reply:
@Arman-Hosseini @PierreRambaud I added a check for `null` like @PierreRambaud suggested.